src/test/java/com/acrolinx/proxy/TomcatIT.java

```diff
@@ -0,0 +1,123 @@
+  private static final StringValuePattern PASSWORD = equalTo("secret");
+  private static final StringValuePattern PROXY_VERSION = equalTo("2");
+  private static final Duration TIMEOUT = Duration.ofSeconds(3);
+  private static final StringValuePattern USERNAME = equalTo("testuser");
+  private static final String X_ACROLINX_CLIENT = "X-Acrolinx-Client";
+
+  private static void checkHttpResponse(HttpResponse<String> httpResponse) {
+    Assertions.assertEquals(200, httpResponse.statusCode());
+    Assertions.assertEquals("{}", httpResponse.body());
+  }
+
+  private static HttpClient createHttpClient() {
+    return HttpClient.newBuilder()
+        .connectTimeout(TIMEOUT)
+        .followRedirects(Redirect.NEVER)
+        .version(Version.HTTP_1_1)
+        .build();
+  }
+
+  private static HttpRequest createHttpRequest(int tomcatPort) throws URISyntaxException {
+    return HttpRequest.newBuilder()
+        .GET()
+        .uri(createUri(tomcatPort))
+        .header(X_ACROLINX_CLIENT, CLIENT_SIGNATURE)
+        .timeout(TIMEOUT)
+        .build();
+  }
+
+  private static URI createUri(int tomcatPort) throws URISyntaxException {
+    return new URI(
+        "http",
+        null,
+        "localhost",
+        tomcatPort,
+        "/acrolinx-proxy-sample/proxy" + URL_STRING,
+        null,
+        null);
+  }
+
+  private static void deployWarFile(Tomcat tomcat) {
+    final Path warFilePath = Path.of("target/acrolinx-proxy-sample.war").toAbsolutePath();
+    Assertions.assertTrue(Files.exists(warFilePath));
+
+    tomcat.addWebapp("", warFilePath.toString());
+  }
+
+  private static void stubWireMock(WireMockServer wireMockServer) {
+    wireMockServer.stubFor(
+        WireMock.request(HTTP_METHOD, urlEqualTo(URL_STRING))
+            .withHeader(X_ACROLINX_CLIENT, equalTo(CLIENT_SIGNATURE))
+            .withHeader("X-Acrolinx-Integration-Proxy-Version", PROXY_VERSION)
+            .withHeader("username", USERNAME)
+            .withHeader("password", PASSWORD)
+            .withRequestBody(AbsentPattern.ABSENT)
+            .willReturn(WireMock.okJson("{}")));
+  }
+
+  private static void verifyInteractionWithWireMock(WireMockServer wireMockServer) {
+    wireMockServer.verify(
+        new RequestPatternBuilder(RequestMethod.fromString(HTTP_METHOD), urlEqualTo(URL_STRING))
+            .withHeader(X_ACROLINX_CLIENT, equalTo(CLIENT_SIGNATURE))
+            .withHeader("X-Acrolinx-Integration-Proxy-Version", PROXY_VERSION)
+            .withHeader("username", USERNAME)
+            .withHeader("password", PASSWORD)
+            .withRequestBody(AbsentPattern.ABSENT));
+
+    Assertions.assertEquals(Collections.emptyList(), wireMockServer.findAllUnmatchedRequests());
+  }
+
+  @Test
+  void embeddedTomcatTest(@TempDir Path tempDirectory) throws Exception {
+    try (WireMockServerWrapper wireMockServerWrapper = WireMockServerWrapper.startOnHttpPort(8031);
```

Comment:
that would be nice, but I don't know how to do it since this port number 8031 needs to match the _acrolinxUrl_ parameter in the `web.xml` file, I have no idea how we could use a random port without breaking the test.

Reply:
I had not realized this. Thanks for clarifying.